transport/src/main/java/io/netty/channel/nio/AbstractNioByteChannel.java

```diff
@@ -208,36 +268,19 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
                 }
             } else if (msg instanceof FileRegion) {
                 FileRegion region = (FileRegion) msg;
-                boolean done = region.transferred() >= region.count();
-
-                if (!done) {
-                    long flushedAmount = 0;
-                    if (writeSpinCount == -1) {
-                        writeSpinCount = config().getWriteSpinCount();
-                    }
-
-                    for (int i = writeSpinCount - 1; i >= 0; i--) {
-                        long localFlushedAmount = doWriteFileRegion(region);
-                        if (localFlushedAmount == 0) {
-                            setOpWrite = true;
-                            break;
-                        }
-
-                        flushedAmount += localFlushedAmount;
-                        if (region.transferred() >= region.count()) {
-                            done = true;
-                            break;
-                        }
-                    }
-
-                    in.progress(flushedAmount);
+                if (writeSpinCount == -1) {
+                    writeSpinCount = config().getWriteSpinCount();
                 }
-
-                if (done) {
-                    in.remove();
-                } else {
-                    // Break the loop and so incompleteWrite(...) is called.
-                    break;
+                switch (doWriteFileRegion(in, region, writeSpinCount)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)